activerecord/test/cases/calculations_test.rb

```diff
@@ -1049,6 +1049,12 @@ def test_ids_with_includes
     assert_equal Company.all.map(&:id).sort, Company.all.includes(:contracts).ids.sort
   end
 
+  def test_ids_with_includes_and_non_primary_key_order
+    rating = 1
+    Company.all.each { |company| company.update!(rating: rating += 1) }
+    assert_equal Company.all.sort_by(&:rating).map(&:id), Company.includes(:comments).order(:rating).ids
```

Comment:
Before the fix:
<img width="1083" alt="Screenshot 2023-05-01 at 9 14 51 am" src="https://user-images.githubusercontent.com/54629302/235408184-5235f577-d96c-4640-a41b-209fbe4007eb.png">